src/main/java/io/xpring/xrpl/model/XrpDepositPreauth.java

```diff
@@ -0,0 +1,80 @@
+package io.xpring.xrpl.model;
+
+import io.xpring.common.XrplNetwork;
+import io.xpring.xrpl.ClassicAddress;
+import io.xpring.xrpl.ImmutableClassicAddress;
+import io.xpring.xrpl.Utils;
+import org.immutables.value.Value;
+import org.xrpl.rpc.v1.DepositPreauth;
+
+import java.util.Optional;
+
+/**
+ * Represents a DepositPreauth transaction on the XRP Ledger.
+ * <p>
+ * A DepositPreauth transaction gives another account pre-approval to deliver payments to the sender of this
+ * transaction.
+ * This is only useful if the sender of this transaction is using (or plans to use) Deposit Authorization.
+ *
+ * Note: authorize and unauthorize are mutually exclusive fields: one but not both should be set.
```

Comment:
I copied https://github.com/xpring-eng/Xpring-JS/pull/493, but...is this sufficient? should we explicitly throw if both are set, or return null, or something?

Reply:
CC @amiecorso

Reply:
you know.... Keefer and I kinda discussed this and decided that:  these classes should always be populated by protobufs that are coming over the wire from the ledger (with the exception of our tests where we obviously construct them by hand).  and we are trusting the ledger not to give us bogus data.  There are other places in these classes that have similar but less obvious mutual exclusion or other rules, and we don't really police things there, either.  However, it does seem possible that while the data on XRPL is probably correct, that the gRPC implementation that's sending things back to us may not have been well-vetted at this point.  So I would say - I'm ok with you just leaving things as-is, but it could be worthwhile to add some additional checks at some point if we have any concerns about the quality of the data.  This is actually a task that I thought about potentially asking Mayuka to do (if she's working on SDK at all - not sure) because it would get her familiar with these classes and also give her a concrete task for getting familiar with some of the workings on XRPL.

Reply:
sounds good. i added https://app.asana.com/0/0/1189602713416473/f to track